core/src/main/java/com/linecorp/armeria/client/retry/RetryingHttpClient.java

```diff
@@ -131,6 +137,9 @@
 
     @Override
     protected HttpResponse doExecute(ClientRequestContext ctx, HttpRequest req) throws Exception {
+        final String authority = req.headers().authority();
+        ctx.attr(AUTHORITY_EXISTS).set(!isNullOrEmpty(authority));
```

Comment:
How about setting only when the authority header exists and checking with `hasAttr()`?

Reply:
Fixed Thank you~!